hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/PrecompileFunctionSelector.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile;
+
+/** All ABI constants used by {@link Precompile} implementations, in one place for easy review. */
+public enum PrecompileFunctionSelector {
```

Comment:
Why do we need to switch to an enum?  The PR that bought this in (#3748) does not provide any explaination as to why this may be a better choice.  For performance I would think passing around and switching on `int`s would be better as it doesn't have JVM native boxing.

Reply:
Pasting the message from Slack for visibility:
```
there was a Sonar issue in the main PR related to the [HTSPrecompiledContract switch](https://github.com/hashgraph/hedera-services/blob/fef7d2290c8b3c00170271083cc6380b73dd874b/hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java#L260) saying more than 30 cases is too large and should be refactored to either use a map or enums.
Since the function selectors are constants and will never change I thought it makes sense to use enums here.
We merged it to enable Sonar checks on the main PR and have a green build. I agree the main PR gets bigger this way, but the alternative is to have 3 PRs related to the same change.
```
@shemnon do you want us to revert commit 064f44169e23215ef11e295e26cdfd8c2aebe5b7 that added the enum?

Reply:
yes please.  That sonar rule looks very arbitrary to me and forces us to add null pointer checks where previously they were not needed.  We can suppress the sonar rule manually or turn the rule off.

Reply:
ok, done